src/states/elder/mod.rs

```diff
@@ -1269,32 +1214,23 @@ impl Elder {
     /// Returns the set of peers that are responsible for collecting signatures to verify a message;
     /// this may contain us or only other nodes. If our signature is not required, this returns
     /// `None`.
-    fn get_signature_targets(&self, src: &SrcLocation) -> Option<BTreeSet<XorName>> {
-        let list: Vec<XorName> = match *src {
-            SrcLocation::Section(pfx) => self
-                .chain
-                .all_sections()
-                .flat_map(|(_, si)| si.member_names())
-                .filter(|name| pfx.matches(name))
-                .copied()
-                .sorted_by(|lhs, rhs| pfx.name().cmp_distance(lhs, rhs)),
-            SrcLocation::Node(_) => {
-                let mut result = BTreeSet::new();
-                let _ = result.insert(*self.name());
-                return Some(result);
-            }
-        };
+    fn get_signature_targets(&self) -> Vec<XorName> {
+        let mut list = self
+            .chain
+            .our_elders()
+            .map(|p2p_node| p2p_node.name())
+            .copied()
+            .sorted_by(|lhs, rhs| self.our_prefix().name().cmp_distance(lhs, rhs));
```

Comment:
Won't it be always the same targets? (won't it always be the first elders (maybe)?) is that what happened before? is that what we want (I thought not).

Reply:
Should it not be the ones closest to the destination?

Reply:
I think the prefix in `SrcLocation::Section` will always be out prefix no? So taking only our section elders should be enough?

> Should it not be the ones closest to the destination?

That's not what we did previously, but maybe it makes sense to do, so we don't always pick the same nodes?

Reply:
> That's not what we did previously, but maybe it makes sense to do, so we don't always pick the same nodes?

That is true, and probably it is not for that PR, or at least a separate commit, it just seem strange that we would always use the same target for message we send.

Reply:
> So taking only our section elders should be enough?
Yes, that is correct, what was strange that we would always take the same of our elders.